sw/device/silicon_creator/lib/ownership/isfb.c

```diff
@@ -0,0 +1,147 @@
+
+  // The expected size of the strike region in bytes and words.
+  kExpectedStrikeRegionBytesCount = kExpectedStrikeBitCount * sizeof(uint32_t),
+  kExpectedStrikeRegionWordsCount =
+      kExpectedStrikeRegionBytesCount / sizeof(uint32_t),
+
+  // The layout of the ISFB info page is as follows:
+  //
+  // | byte start      | byte end | description
+  // |-----------------|----------|------------
+  // | 0               | 511      | Strike mask
+  // | 512             | 1023     | Unused and available to owner firmware.
+  // | 1024            | OnwerCfg | Product expression constraints.
+  // | 1024 + OwnerCfg | 2047     | Unused and available to owner firmware.
+  kIsfbWordOffsetStartStrikeRegion = 0,
+  kIsfbWordOffsetStartProductExpr = 1024 / sizeof(uint32_t),
+};
+static_assert(kExpectedStrikeRegionWordsCount == kExpectedStrikeBitCount,
+              "Unexpected strike region size");
+static_assert(kIsfbWordOffsetStartProductExpr * sizeof(uint32_t) +
+                      KProductExprMaxCountBytes ==
+                  kFlashPageBytes,
+              "Product expression total size overflows page size");
+
+rom_error_t isfb_boot_request_process(const manifest_ext_isfb_t *ext,
+                                      const owner_config_t *owner_config,
+                                      uint32_t *checks_performed_count) {
+  *checks_performed_count = UINT32_MAX;
+  if (ext->header.name != kManifestExtIdIsfb ||
+      (hardened_bool_t)owner_config->isfb == kHardenedBoolFalse) {
+    return kErrorOwnershipISFBNotPresent;
+  }
+  const owner_isfb_config_t *isfb = owner_config->isfb;
+  if (ext->product_expr_count * 2 > isfb->product_words ||
+      ext->product_expr_count > kProductExprMaxCount) {
+    return kErrorOwnershipISFBProductExpCnt;
+  }
+
+  flash_ctrl_perms_t perm = {
+      .read = kMultiBitBool4True,
+      .write = kMultiBitBool4True,
+      .erase = kMultiBitBool4False,
+  };
+  flash_ctrl_cfg_t cfg = {
+      .scrambling = kMultiBitBool4False,
+      .ecc = kMultiBitBool4True,
+      .he = kMultiBitBool4False,
+  };
+
+  flash_ctrl_info_page_t isfb_info_page;
+  HARDENED_RETURN_IF_ERROR(flash_ctrl_info_type0_params_build(
+      isfb->bank, isfb->page, &isfb_info_page));
+  flash_ctrl_info_perms_set(&isfb_info_page, perm);
+  flash_ctrl_info_cfg_set(&isfb_info_page, cfg);
+
+  // There are in total 128 bits in the strike mask. Each bit corresponds to a
+  // flash word. Each flash word is 64 bits.
+  uint32_t data[kFlashPageWords];
+  static_assert(sizeof(ext->strike_mask) * CHAR_BIT == kExpectedStrikeBitCount,
+                "Strike mask size mismatch");
+  static_assert(sizeof(data) >= kFlashPageBytes, "Unexpected data size");
+
+  HARDENED_RETURN_IF_ERROR(flash_ctrl_info_read(&isfb_info_page, /*offset=*/0,
+                                                ARRAYSIZE(data), data));
+
+  uint32_t strike_cnt_ok = 0;
+  uint32_t strike_cnt_bad = 0;
+  uint32_t *strike_word = &data[kIsfbWordOffsetStartStrikeRegion];
```

Comment:
Since the strike words correspond to flash words, should this be `uint64_t` instead? (and use UINT64_MAX in the tests below)?

Reply:
Updated the documentation to be consistent with the implementation. We switched to 32bit strike words to compress the anti-rollback region. 

https://github.com/lowRISC/opentitan/pull/25147#discussion_r1844160952